src/bin/pgcopydb/cli_list.c

```diff
@@ -1326,6 +1317,13 @@ cli_list_schema(int argc, char **argv)
 		/* errors have already been logged */
 		exit(EXIT_CODE_BAD_ARGS);
 	}
+	/* allocating it to the length of listDBoptions.source_pguri as ListDBOptions does not have target_pguri */
+	options.target_pguri = calloc(strlen(listDBoptions.source_pguri)+1, sizeof(char));
+	if(options.target_pguri == NULL)
+	{
+		log_error(ALLOCATION_FAILED_ERROR);
+		exit(EXIT_CODE_INTERNAL_ERROR);
+	}
```

Comment:
In the function copydb_init_specs_from_listdboptions() I see that we just keep the pointer around, so I don't think we need to allocate memory here. If we want to duplicate the memory, I would ask to use strdup() anyway, within the function cited above.

Reply:
But pointing `options->target_pguri` to `listDBoptions.source_pguri` or using `strdup` wont be fully correct either, right? Even it has to be done, shouldnt it be done in `cli_list_db_getopts` and not in `copydb_init_specs_from_listdboptions` because that function is only called for cli_list commands which need the source URI to be set. 
Anyway , we still will have allocate this memory , because the NULL checks I added in `copydb.c` would fail.
But, we cant even remove those null checks because `copydb_init_specs` is called at multiple other places which require  the target_uri to not be NULL.